src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriverAdapter.java

```diff
@@ -3,28 +3,38 @@
 import edu.kis.legacy.drawer.panel.DrawPanelController;
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.macros.DriverCallRecorder;
+import edu.kis.powp.jobs2d.macros.DriverCallRecorderObservable;
+import edu.kis.powp.jobs2d.macros.DriverCallRecorderObserver;
 
 /**
  * Line adapter - Job2dDriver with DrawPanelController object.
  */
-public class LineDriverAdapter implements Job2dDriver {
+public class LineDriverAdapter extends DriverCallRecorderObservable implements Job2dDriver {
```

Comment:
Instead of brutally extending LineDriverAdapter please use SRP (e.g., decorator or composite pattern in drivers hierarchy)